issuance/issuance.go

```diff
@@ -650,12 +658,17 @@ func (i *Issuer) Prepare(req *IssuanceRequest) ([]byte, *issuanceToken, error) {
 
 	if req.IncludeCTPoison {
 		template.ExtraExtensions = append(template.ExtraExtensions, ctPoisonExt)
-	} else if req.SCTList != nil {
-		sctListExt, err := generateSCTListExt(req.SCTList)
+	} else if len(req.sctList) > 0 {
+		if len(req.precertDER) == 0 {
+			return nil, nil, fmt.Errorf("inconsistent request contains SCTList but no PrecertDER")
+		}
+		sctListExt, err := generateSCTListExt(req.sctList)
 		if err != nil {
 			return nil, nil, err
 		}
 		template.ExtraExtensions = append(template.ExtraExtensions, sctListExt)
+	} else {
+		return nil, nil, fmt.Errorf("invalid request contains neither SCTList nor PrecertDER")
```

Comment:
Case nit and use of `fmt.Errorf` without formatting directives.

```suggestion
		return nil, nil, errors.New("invalid request contains neither sctList nor precertDER")
```